### PR TITLE
Refact core list test

### PR DESCRIPTION
**Pull request by fredbi**
Following a remark from @kerneltime, modified tests / mocks in core to abide by path conventions adopted by the model package. This unlocked the ability to implement more sanity checks on paths in model.

**NOTE**: thinking more about all this (and trying to figure out how to document it), I feel like all the remote config organization of files should be delegated to a subpackage, like `model/paths`.

### CODE REVIEW

File: pkg/core/bundle_pack.go [hunk 1]
```diff
@@ -297,7 +297,7 @@ func uploadBundle(ctx context.Context, bundle *Bundle, bundleEntriesPerFile uint
 	}
 	bundle.l.Info("uploaded filelists",
 		zap.Int("actual number uploads attempted", numFileListUploads),
-		zap.Int("approx expected number of uploads", numFilePackedRes/int(bundleEntriesPerFile)),
+		zap.Int("approx expected number of uploads", maxInt(numFilePackedRes/int(bundleEntriesPerFile), 1)),
```
Comment (fredbi) [inline pkg/core/bundle_pack.go]: small extra fix: avoids approximation to be zero for low numbers, but 1.

File: pkg/model/paths.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+const (
+	// descriptor files (object metadata)
+
+	repoDescriptorFile    = "repo.yaml"
+	labelDescriptorFile   = "label.yaml"
+	bundleDescriptorFile  = "bundle.yaml"
+	contextDescriptorFile = "context.yaml"
+	bundleFilesIndex      = "bundle-files-"
```
Comment (kerneltime) [inline pkg/model/paths.go]: bundleFilesIndex => bundleFilesIndexPrefix?

Comment (fredbi) [reply pkg/model/paths.go]: Yeah ok

**Comment by ransomw1c** [comment]
cool! the `GetArchivePathComponents` cast lives on..

**Pull request merged**